nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/impl/TestPropertyCommentManager.java

```diff
@@ -625,13 +641,66 @@ public void testGetCommentAsRegularUser() {
             assertEquals("The user jane does not have access to the comments of document " + doc.getId(),
                     e.getMessage());
         }
+
+        try (CloseableCoreSession janeSession = CoreInstance.openCoreSession(doc.getRepositoryName(), "jane")) {
+            DocumentModelList documents = janeSession.getDocuments(
+                    new DocumentRef[] { new IdRef(comment.getId()), new IdRef(subComment.getId()) });
+            assertEquals(0, documents.size());
+        }
+    }
+
+    @Test
+    public void testQueryCommentAsRegularUser() {
+
+        DocumentModel domain = session.createDocumentModel("/", "domain", "Domain");
+        session.createDocument(domain);
+        DocumentModel doc = session.createDocumentModel("/domain", "test", "File");
+        doc = session.createDocument(doc);
+        ACPImpl acp = new ACPImpl();
+        ACL acl = acp.getOrCreateACL();
+        acl.add(new ACE("john", SecurityConstants.READ, true));
+        // acl.add(new ACE("jane", SecurityConstants.BROWSE, true));
+        session.setACP(doc.getRef(), acp, false);
+        session.save();
+
+        String query = String.format("Select * From Document where %s = '%s'", NXQL.ECM_PARENTID, doc.getId());
+
+        try (CloseableCoreSession johnSession = CoreInstance.openCoreSession(doc.getRepositoryName(), "john")) {
+            String author = "john";
+            String text = "I am a comment !";
+            Comment comment = new CommentImpl();
+            comment.setAuthor(author);
+            comment.setText(text);
+            comment.setParentId(doc.getId());
+            comment = commentManager.createComment(johnSession, comment);
+
+            Comment subComment = new CommentImpl();
+            subComment.setAuthor(author);
+            subComment.setText("I reply to the last comment");
+            subComment.setParentId(comment.getId());
+            commentManager.createComment(johnSession, subComment);
+
+            johnSession.save();
+
+            DocumentModelList documents = johnSession.query(query);
+            assertEquals(2, documents.size());
+
+            assertEquals(doc.getId(), johnSession.getDocument(documents.get(0).getParentRef()).getId());
+            assertEquals(doc.getId(), johnSession.getDocument(documents.get(1).getParentRef()).getId());
+        }
+
+        // FIXME understand Browse Permission
```

Comment:
What's the question?

Reply:
Explenations about the difference between read and browse permission are welcome. I feel it's still unclear for me

Reply:
because if we give the `Browse` permission to `jane` as ` acl.add(new ACE("jane", SecurityConstants.BROWSE, true));` our test failed, we expected that this permission works differently then the `READ` one

`CoreInstance.openCoreSession(doc.getRepositoryName(), "jane")` -> will get the doc with Browse permission.